nengo/solvers.py

```diff
@@ -67,16 +72,14 @@ def mul_encoders(self, Y, E, copy=False):
         copy : bool, optional (Default: False)
             Whether a copy of ``Y`` should be returned if ``E`` is None.
         """
-        if self.weights:
-            if E is None:
-                raise ValidationError(
-                    "Encoders must be provided for weight solver", attr='E')
-            return np.dot(Y, E)
-        else:
-            if E is not None:
-                raise ValidationError(
-                    "Encoders must be 'None' for decoder solver", attr='E')
-            return Y.copy() if copy else Y
+        if self.weights and E is None:
+            raise ValidationError(
+                "Encoders must be provided for weight solver", attr='E')
+        if not self.weights and E is not None:
+            raise ValidationError(
+                "Encoders must be 'None' for decoder solver", attr='E')
+
+        return np.dot(Y, E) if E is not None else Y.copy() if copy else Y
```

Comment:
Nah, not a critical thing, I see your point! I think for me, I parse these lines with an eye to the value, rather than the condition... so when I see `x if a else y if b else z if c else q` I'm thinking about what is the situation in which you get `b`, rather than reading it left to right. So I'm looking at `q` for example, then I scan left and see that `q` happens if `a` `b` and `c` are False, which takes a few seconds to think through. For some reason the parens help with the mental organization of that, but like I said, not a critical thing so no need to change.